### PR TITLE
[WIP] networking: Basic networking using flannel

### PR DESCRIPTION
**Pull request by jellonek**
As in subject.
TODO:
* [x] compute addressing from subnet
* [x] fail (temporarily, failback to next driver (planned calico support) in future) if FLANNEL_SUBNET is empty
* [x] get network info about running "container" from libvirt and store it in sandbox

This change is [Reviewable](https://reviewable.io/reviews/mirantis/virtlet/17)

### CODE REVIEW

**Comment by jellonek** [comment]
@nhlfr ptal

File: pkg/libvirttools/networking.c [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+		}
+	}
+exit:
+	return;
+}
+
+int
+getDomIfAddr(virConnectPtr conn, char *uuid, char **ip)
+{
+        int result = 0;
```
Comment (nhlfr) [inline pkg/libvirttools/networking.c]: https://www.youtube.com/watch?v=SsoOG6ZeyUI

File: pkg/libvirttools/networking.c [hunk 2]
```diff
@@ -0,0 +1,111 @@
...
+		if (!iface->naddrs) {
+			continue;
+		}
+
+		for (j = 0; j < iface->naddrs; j++) {
+			if (iface->addrs[j].type != VIR_IP_ADDR_TYPE_IPV4) {
+				continue;
+			}
+			*ip = strdup(iface->addrs[j].addr);
+			goto exit;
```
Comment (nhlfr) [inline pkg/libvirttools/networking.c]: If we are not going to free any pointer in this function, I think we may just `return` without `goto`.

File: pkg/libvirttools/networking.c (hunk 1 above)
Comment (pigmej) [inline pkg/libvirttools/networking.c]: @nhlfr to avoid that we could plug codeclimate there.

File: pkg/libvirttools/networking.c (hunk 2 above)
Comment (vefimova) [inline pkg/libvirttools/networking.c]: It's better to use `break` here, as we we'll have function end at the common place.

File: pkg/libvirttools/networking.go [hunk 3]
```diff
@@ -53,7 +102,10 @@ func generateNetworkXML(subnet string, device string) string {
 	</dhcp>
     </ip>
 </network>`
-	return fmt.Sprintf(xml, defaultName, defaultName, defaultBridge, device, address, netmask, rangeStart, rangeEnd)
+
+	return fmt.Sprintf(
+		xml, defaultName, defaultName, defaultBridge, iface,
```
Comment (vefimova) [inline pkg/libvirttools/networking.go]: Looks like dup of defaultName.

Comment (jellonek) [inline pkg/libvirttools/networking.go]: Yep, you are right!

File: pkg/libvirttools/networking.c (hunk 1 above)
Comment (jellonek) [inline pkg/libvirttools/networking.c]: I hate tabs... Thx.

File: pkg/libvirttools/networking.c (hunk 2 above)
Comment (jellonek) [inline pkg/libvirttools/networking.c]: Hmmm. Both seems to be good for me (single point of exit rule vs no-goto).
I'll change this to `return`.

Comment (jellonek) [inline pkg/libvirttools/networking.c]: Done.

File: pkg/libvirttools/networking.c (hunk 1 above)
Comment (jellonek) [inline pkg/libvirttools/networking.c]: Done.

File: pkg/libvirttools/networking.go (hunk 3 above)
Comment (jellonek) [inline pkg/libvirttools/networking.go]: Done.

**Comment by jellonek** [comment]
Have to abandon this.

**Pull request closed without merging**